### PR TITLE
Merge svn427

### PR DESCRIPTION
**Pull request by ligenxxxx**
Need to test.

### CODE REVIEW

**Comment by SumolX** [comment]
I will test this PR tomorrow and provide feedback... in regards to the functionality as I fly mostly hdzero and not analog.

**Comment by SumolX** [comment]
Fixed Emulator Right Click/Press detection:
https://github.com/hd-zero/hdzero-goggle/commit/9b4c78b498f48e4566ed6d4d11655125fcf33f73

**Comment by SumolX** [comment]
**Restructured keyboard/UI Input Handling:**
https://github.com/hd-zero/hdzero-goggle/commit/a39623670d7bc27e82cf5dfd99b6dea079554285

**When keyboard is not active and we are highlighting an input text field:**

dial click will display the keyboard.

**When the keyboard is actively displayed:**

dial click will simulate the press of the highlighted key.
right press will clear keyboard text field.
right click will close the keyboard and update input text field.

Per our conversation this aligns with how users have become accustom to manipulating the UI.

File: src/ui/page_connections.c
Comment (pkendall64) [inline src/ui/page_connections.c:1]: This file is empty and should not be added to the PR

**Pull request merged**